File: Python/ЕГЭ2021;2022/24/Решение заданий. Поляков/4753.py
'''
Текстовый файл 24-181.txt содержит строку из заглавных латинских букв и точек, 
всего не более 106 символов. Определите максимальное количество идущих подряд символов, 
среди которых нет гласных букв (символов A, E, I, O, U, Y), но есть не менее 6 точек.
'''
# https://prnt.sc/KOkYp2thVnFm


f = open('24.txt')
s = f.readline()


elements = ['A', 'E', 'I', 'O', 'U', 'Y']

for i in elements:
    s = s.replace(i, ' ')


max_strin = 0


for line in s.split():
    if line.count('.') >= 6:
        max_strin = max(max_strin, len(line))


print(max_strin)

File: C++/ЕГЭ2021;2022/24/Решение заданий. Поляков/4753.cpp
#include<bits/stdc++.h>
#include<regex>
#define istr istream_iterator
using namespace std;
int main(){
    fstream f; f.open("181.txt");
    string s; f >> s;
    vector<string> p{"A", "E", "I", "O", "U", "Y"};
    for(auto&c:p){
        s=regex_replace(s, regex(c), " ");
    }
    istringstream iss(s);
    vector<string>sub(istr<string>{iss}, istr<string>());
    int res=0;
    for(auto&s:sub){
        if(count(s.begin(), s.end(), '.')>=6) res=max(res, int(s.size())); 
    }
    cout << res;
}